src/test/ui/self/elision/ref-alias.rs

```diff
@@ -0,0 +1,39 @@
+// run-pass
+
+#![feature(arbitrary_self_types)]
+#![allow(non_snake_case)]
+
+use std::pin::Pin;
+
+struct Struct { }
+
+type Alias = Struct;
+
+impl Struct {
+    // Test using an alias for `Struct`:
+    //
+    // FIXME. We currently fail to recognize this as the self type, which
+    // feels like a bug.
```

Comment:
👍 -- If `self: &Struct` is OK then so should this!

Reply:
Related: https://github.com/rust-lang/rust/pull/60944#issuecomment-495346120 (#60944 unintentionally tried to support this.)

Reply:
See comment in https://github.com/rust-lang/rust/pull/61207#discussion_r305597812.